plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcCachingConnectorSmokeTest.java

```diff
@@ -32,11 +38,52 @@ protected QueryRunner createQueryRunner()
                 .putAll(TestingH2JdbcModule.createProperties())
                 .put("metadata.cache-ttl", "10m")
                 .put("metadata.cache-missing", "true")
+                .put("case-insensitive-name-matching", "true")
                 .put("allow-drop-table", "true")
                 .build();
+        this.h2SqlExecutor = new JdbcSqlExecutor(properties.get("connection-url"), new Properties());
         return createH2QueryRunner(REQUIRED_TPCH_TABLES, properties);
     }
 
+    @Test
+    public void testFlushMetadataCacheProcedureFlushMetadata()
+    {
+        assertUpdate("CREATE TABLE cached AS SELECT * FROM nation", 25);
+        assertQuerySucceeds("SELECT name, regionkey FROM cached");
+
+        // Rename column outside Trino
+        h2SqlExecutor.execute("ALTER TABLE tpch.cached ALTER COLUMN regionkey RENAME TO renamed");
+
+        String renamedColumnQuery = "SELECT name, renamed FROM cached";
+        // Should fail as Trino has old metadata cached
+        assertThatThrownBy(() -> getQueryRunner().execute(renamedColumnQuery))
+                .hasMessageMatching(".*Column 'renamed' cannot be resolved");
+
+        // Should succeed after flushing Trino JDBC metadata cache
+        getQueryRunner().execute(getSession(), "CALL system.flush_metadata_cache()");
+        assertQuerySucceeds(renamedColumnQuery);
+    }
+
+    @Test
+    public void testFlushMetadataCacheProcedureFlushIdentifierMapping()
+    {
+        assertUpdate("CREATE TABLE cached_name AS SELECT * FROM nation", 25);
+
+        String query = "SELECT name, regionkey FROM cached_name";
+        assertQuerySucceeds(query);
+
+        // Rename to lowercase name outside Trino (H2 stores unquoted names as uppercase)
+        h2SqlExecutor.execute("ALTER TABLE tpch.CACHED_NAME RENAME TO tpch.\"cached_name\"");
```

Comment:
Since we used quoted name for the target table we should quote the original name too to make it clearer that the original table had an uppercase name.

```suggestion
        h2SqlExecutor.execute("ALTER TABLE tpch.\"CACHED_NAME\" RENAME TO tpch.\"cached_name\"");
```

Another alternative is to `verify` our assumption that the table has an uppercase name - but doesn't seem worth it (since we need to add a method that returns result set from H2 queries).

Reply:
I've added verify that actual table stored in H2 is uppercase by checking if query for lowercase fails and uppercase succeed's .